api/v1alpha2/zz_generated.conversion.go

```diff
@@ -603,6 +603,7 @@ func autoConvert_v1alpha3_AWSMachineSpec_To_v1alpha2_AWSMachineSpec(in *v1alpha3
 	}
 	// WARNING: in.RootVolume requires manual conversion: does not exist in peer-type
 	out.NetworkInterfaces = *(*[]string)(unsafe.Pointer(&in.NetworkInterfaces))
+	// WARNING: in.UncompressedUserData requires manual conversion: does not exist in peer-type
```

Comment:
This will require a modification to `api/v1alpha2/awscluster_conversion.go` to attempt to restore the value from annotation if it is set, similar to how `Status.Bastion` is handled here: https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/7240d095ea189681c6f92c79e14dd37853af2512/api/v1alpha2/awscluster_conversion.go#L65-L67

Reply:
I added a unit test in awsmachine_conversion_test.go. 

https://github.com/minsheng-fintech-corp-ltd/cluster-api-provider-aws/blob/35ec7d2a453c10ab1ce5ddca85ccaa7744f4eb2e/api/v1alpha2/awsmachine_conversion_test.go#L154-L166

and added manual conversion in restoreAWSMachineSpec
 
https://github.com/minsheng-fintech-corp-ltd/cluster-api-provider-aws/blob/master/api/v1alpha2/awsmachine_conversion.go

Thanks for the info!